ngraph/test/backend/binary_convolution.in.cpp

```diff
@@ -0,0 +1,201 @@
+                            const std::vector<T_IN>& filters,
+                            const Shape filters_shape,
+                            const std::vector<T_IN>& outputs,
+                            const Shape outputs_shape,
+                            const Strides& strides,
+                            const CoordinateDiff& padding,
+                            const Strides& dilations)
+{
+    const CoordinateDiff pads_begin{padding};
+    const CoordinateDiff pads_end{padding};
+    const op::PadType auto_pad{op::PadType::EXPLICIT};
+
+    auto inputs_param = make_shared<op::Parameter>(element::from<T_IN>(), inputs_shape);
+    auto filters_param = make_shared<op::Parameter>(element::from<T_IN>(), filters_shape);
+    auto conv = make_shared<op::v1::Convolution>(
+        inputs_param, filters_param, strides, pads_begin, pads_end, dilations, auto_pad);
+    auto f = make_shared<Function>(conv, ParameterVector{inputs_param, filters_param});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    auto input_tensor = backend->create_tensor(element::from<T_IN>(), inputs_shape);
+    copy_data(input_tensor, inputs);
+    auto filters_tensor = backend->create_tensor(element::from<T_IN>(), filters_shape);
+    copy_data(filters_tensor, filters);
+    auto result = backend->create_tensor(element::from<T_IN>(), outputs_shape);
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result}, {input_tensor, filters_tensor});
+    EXPECT_TRUE(test::all_close_f((outputs), read_vector<T_IN>(result), MIN_FLOAT_TOLERANCE_BITS));
+}
+
+// clang-format off
+// --------------------- 1D convolution ------------------------------------------
+NGRAPH_TEST(${BACKEND_NAME}, bin_convolution_1D_1batch_1channel_no_padding)
+{
+    const Strides strides{1};
+    const CoordinateDiff padding{0};
+    const Strides dilations{1};
+
+    const Shape inputs_shape{1, 1, 5};
+    const std::vector<float> inputs_conv{1.0f, -1.0f, -1.0f, 1.0f, -1.0f};
+    const std::vector<float> inputs_bin_conv{1.0f, 0.0f, 0.0f, 1.0f, 0.0f};
+
+    const Shape filters_shape{1, 1, 3};
+    const std::vector<float> filters_conv{1.0f, -1.0f, 1.0f};
+    const std::vector<uint8_t> filters_bin_conv{160}; // filters 1D: {1.0f, 0.0f, 1.0f}
+
+    const Shape outputs_shape{1, 1, 3};
+    const std::vector<float> outputs{1.0f, 1.0f, -3.0f};
+
+    BinaryConvolutionTest(
+        inputs_bin_conv,
+        inputs_shape,
+        filters_bin_conv,
+        filters_shape,
+        outputs,
+        outputs_shape,
+        strides,
+        padding,
+        dilations);
+
+    ConvolutionTest(
+        inputs_conv,
+        inputs_shape,
+        filters_conv,
+        filters_shape,
+        outputs,
+        outputs_shape,
+        strides,
+        padding,
+        dilations);
+}
+
+// --------------------- 2D convolution ------------------------------------------
+NGRAPH_TEST(${BACKEND_NAME}, bin_convolution_2D_1batch_1channel_no_padding)
```

Comment:
Please add more 2D test cases with stride, padding, dilation, multiple channels, filters and batches (follow what was done in Convolution) and also cover new pad_value attribute.

Reply:
Done